mongo/options/clientoptions.go

```diff
@@ -300,187 +322,305 @@ func (c *ClientOptions) ApplyURI(uri string) *ClientOptions {
+// "connectTimeoutMS" (e.g "connectTimeoutMS=30") option. If set to 0, no timeout will be used. The default is 30
+// seconds.
 func (c *ClientOptions) SetConnectTimeout(d time.Duration) *ClientOptions {
 	c.ConnectTimeout = &d
 	return c
 }
 
-// SetDialer specifies a custom dialer used to dial new connections to a server.
-// If a custom dialer is not set, a net.Dialer with a 300 second keepalive time will be used by default.
+// SetDialer specifies a custom ContextDialer to be used to create new connections to the server. The default is a
+// net.Dialer instance with a 300 second keepalive time.
 func (c *ClientOptions) SetDialer(d ContextDialer) *ClientOptions {
 	c.Dialer = d
 	return c
 }
 
-// SetDirect specifies whether the driver should connect directly to the server instead of
-// auto-discovering other servers in the cluster.
+// SetDirect specifies whether or not a direct connect should be made. To use this option, a URI with a single host must
+// be specified through ApplyURI. If set to true, the driver will only connect to the host provided in the URI and will
+// not discover other hosts in the cluster. This can also be set through the "connect" URI option with the following
+// values:
+//
+// 1. "connect=direct" for direct connections
+//
+// 2. "connect=automatic" for automatic discovery.
+//
+// The default is false.
 func (c *ClientOptions) SetDirect(b bool) *ClientOptions {
 	c.Direct = &b
 	return c
 }
 
-// SetHeartbeatInterval specifies the interval to wait between server monitoring checks.
+// SetHeartbeatInterval specifies the amount of time to wait between periodic background server checks. This can also be
+// set through the "heartbeatIntervalMS" URI option (e.g. "heartbeatIntervalMS=10000"). The default is 10 seconds.
 func (c *ClientOptions) SetHeartbeatInterval(d time.Duration) *ClientOptions {
 	c.HeartbeatInterval = &d
 	return c
 }
 
-// SetHosts specifies the initial list of addresses from which to discover the rest of the cluster.
+// SetHosts specifies a list of host names or IP addresses for servers in a cluster. Both IPv4 and IPv6 addresses are
+// supported. IPv6 literals must be enclosed in '[]' following RFC2732 syntax.
+//
+// Hosts can also be specified as a comma-separated list in a URI. For example, to include "localhost:27017" and
+// "localhost:27018", a URI could be "mongodb://localhost:27017,localhost:27018". The default is ["localhost:27017"]
 func (c *ClientOptions) SetHosts(s []string) *ClientOptions {
 	c.Hosts = s
 	return c
 }
 
-// SetLocalThreshold specifies how far to distribute queries, beyond the server with the fastest
-// round-trip time. If a server's roundtrip time is more than LocalThreshold slower than the
-// the fastest, the driver will not send queries to that server.
+// SetLocalThreshold specifies the width of the 'latency window': when choosing between multiple suitable servers for an
```

Comment:
"width" is a bit of a misnomer.  Did you get this from PyMongo?  Can you canvas other drivers/spec for a better description?

Reply:
I've never heard width used this way, but I can live with it.